features/step_definitions/admin_node/packages_steps.rb

```diff
@@ -9,6 +9,13 @@
 end
 
 Given(/^all dependencies of installed packages are satisfied$/) do
-  admin_node.exec!("zypper", "verify --no-recommends")
+  wait_for "zypper verify --no-recommends", max: "2 minutes", sleep: "5 seconds" do
+    result = admin_node.exec!("zypper", "verify --no-recommends", capture_error: true)
+    if result.exit_code == 7 # ZYPPER_EXIT_ZYPP_LOCKED
+      log.warn(result.error)
+      next
+    end
+    break if result.exit_code.zero?
```

Comment:
why breaking only on exit_code.zero? it looks like you#re ignoring exit codes != 0 && != 7 (e.g. exit 1 for error)